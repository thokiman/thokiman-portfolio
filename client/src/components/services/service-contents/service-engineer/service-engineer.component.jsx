import React from "react";
import { connect } from "react-redux";
import { CSSTransitionGroup } from "react-transition-group";
import { createStructuredSelector } from "reselect";
import { selectIsSideBarHidden } from "../../../../redux/header/header.selectors";
import { selectEngineerService } from "../../../../redux/service/service.selectors";
import ServiceGeneralContent from "../../service-commons/service-general-content/service-general-content.component";
import ServiceIconContent from "../../service-commons/service-icon-content/service-icon-content.component";
import {
  RightServiceContainerEng,
  RightServiceContentEng
} from "./service-enginner.styles";


export const ServiceEngineer = ({
  isSideBarHidden,
  engineer: { title, description, items },
}) => {
  return (
    <RightServiceContainerEng $issidebarhidden={isSideBarHidden}>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={800}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <RightServiceContentEng>
          <ServiceGeneralContent title={title} description={description} />
          <ServiceIconContent items={items} />
        </RightServiceContentEng>
      </CSSTransitionGroup>
    </RightServiceContainerEng>
  );
};

const mapStateToProps = createStructuredSelector({
  engineer: selectEngineerService,
  isSideBarHidden: selectIsSideBarHidden,
});
export default connect(mapStateToProps)(ServiceEngineer);
