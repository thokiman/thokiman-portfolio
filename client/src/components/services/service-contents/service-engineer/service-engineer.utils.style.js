import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../../utils/common.component.utils';

export const measureRightPageContainerEngProps = (viewWidth, viewHeight) => {
  const rightServiceContainerEngHeightOffset = 0;
  const rightServiceContainerEngTopOffset = 0;

  let rightServiceContainerProps;

  if (hasScreenSize1(viewWidth)) {
    console.log('service-engineer', 1);
    rightServiceContainerProps = {
      height: 205 + rightServiceContainerEngHeightOffset,
      top: -60 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('service-engineer', 2);
    rightServiceContainerProps = {
      height: 205 + rightServiceContainerEngHeightOffset,
      top: -40 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('service-engineer', 3);
    rightServiceContainerProps = {
      height: 150 + rightServiceContainerEngHeightOffset,
      top: -50 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('service-engineer', 4);
    rightServiceContainerProps = {
      height: 140 + rightServiceContainerEngHeightOffset,
      top: -50 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('service-engineer', 5);
    rightServiceContainerProps = {
      height: 170 + rightServiceContainerEngHeightOffset,
      top: -50 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('service-engineer', 6);
    rightServiceContainerProps = {
      height: 180 + rightServiceContainerEngHeightOffset,
      top: -50 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('service-engineer', 7);
    rightServiceContainerProps = {
      height: 170 + rightServiceContainerEngHeightOffset,
      top: 0 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('service-engineer', 8);
    rightServiceContainerProps = {
      height: 195 + rightServiceContainerEngHeightOffset,
      top: -40 + rightServiceContainerEngTopOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('service-engineer', 9);
    rightServiceContainerProps = {
      height: 200 + rightServiceContainerEngHeightOffset,
      top: -50 + rightServiceContainerEngTopOffset,
    };
  } else {
    console.log('service-engineer', 10);
    rightServiceContainerProps = {
      height: 245,
      top: -50,
    };
  }

  return rightServiceContainerProps;
};
