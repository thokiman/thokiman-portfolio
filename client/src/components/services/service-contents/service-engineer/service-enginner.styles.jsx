import styled, { css } from 'styled-components';

const backgroundColorRightServiceContainerEng = 'rgba(18, 129, 148, 0.75)';
const maxWidthSmartphone = 800;

const ServiceEngineerIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebarhidden }) => {
  if ($issidebarhidden) return ServiceEngineerIsNotActivated;
  return null;
};

export const RightServiceContainerEng = styled.div`
  background-color: ${backgroundColorRightServiceContainerEng};
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    overflow-y: hidden;
    width: 100%;
    right: 0;
    height: 205vh;

    ${getSideBarActive}
  }
  @media only screen and (max-width: 550px) {
    height: 205vh;
    ${getSideBarActive}
  }
  @media only screen and (max-width: 470px) and (max-height: 823px) {
    height: 140vh;
    ${getSideBarActive}
  }
  @media only screen and (max-width: 470px) and (max-height: 812px) {
  }
  @media only screen and (max-width: 470px) and (max-height: 736px) {
    height: 168vh;
    ${getSideBarActive}
  }
  @media only screen and (max-width: 470px) and (max-height: 667px) {
    height: 170vh;
    ${getSideBarActive}
  }
  @media only screen and (max-width: 350px) and (max-height: 568px) {
    height: 183vh;
    ${getSideBarActive}
  }
  @media only screen and (max-width: 300px) {
    height: 170vh;
    ${getSideBarActive}
  }
`;

export const RightServiceContentEng = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerEng.displayName = 'RightServiceContainerEng';
RightServiceContentEng.displayName = 'RightServiceContentEng';
