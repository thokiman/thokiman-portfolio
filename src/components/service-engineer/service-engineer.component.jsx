import React from "react";

import ServiceGeneralContent from "../service-general-content/service-general-content.component";
import ServiceIconContent from "../service-icon-content/service-icon-content.component";
import "./service-engineer.styles.scss";

const ServiceEngineer = ({ engineer: { title, description, items } }) => {
  return (
    <div className="right-service-container-eng">
      <div className="right-service-content-eng">
        <ServiceGeneralContent title={title} description={description} />
        <ServiceIconContent items={items} />
      </div>
    </div>
  );
};

export default ServiceEngineer;
