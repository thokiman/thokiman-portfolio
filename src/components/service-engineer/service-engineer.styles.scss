.right-service-container-eng {
    background-color: rgba($color: #128194, $alpha: 0.75);
    width: 69.9%;
    height: 94%;
    position: relative;
    right: 40px;
    padding: 10px 0;
    overflow-y: scroll;
    overflow-x: hidden;

    .right-service-content-eng {
        width: 95%;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: flex-start;
        margin: auto 0;
        padding: 20px;
    }

}