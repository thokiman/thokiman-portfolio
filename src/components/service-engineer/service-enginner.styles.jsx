import styled from "styled-components";

export const RightServiceContainerEng = styled.div`
  background-color: rgba(18, 129, 148, 0.75);
  width: 66.3vw;
  height: 68vh;
  padding: 10px 20px;
  position: relative;
  right: 4vw;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
`;

export const RightServiceContentEng = styled.div`
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  margin: auto 0;
  padding: 20px;
`;

RightServiceContainerEng.displayName = "RightServiceContainerEng";
RightServiceContentEng.displayName = "RightServiceContentEng";
