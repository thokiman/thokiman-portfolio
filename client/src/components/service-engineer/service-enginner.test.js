import React from "react";
import { shallow } from "enzyme";

import { ServiceEngineer } from "./service-engineer.component";

describe("ServiceEngineer   component", () => {
  let wrapper;

  const mockFalseBool = false;
  const mockText = "";
  const mockArray = [];
  beforeEach(() => {
    const mockProps = {
      isSideBarHidden: mockFalseBool,
      engineer: { title: mockText, description: mockText, items: mockArray },
    };
    wrapper = shallow(<ServiceEngineer {...mockProps} />);
  });

  it("should render ServiceEngineer   component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
