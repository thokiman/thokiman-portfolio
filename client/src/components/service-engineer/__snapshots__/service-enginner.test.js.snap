// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceEngineer   component should render ServiceEngineer   component 1`] = `ShallowWrapper {}`;
